[teojunda] iP

teojunda opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen [Source](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html)

DukePro frees your mind of having to remember things you need to do. It's,
+ text-based
+ easy to learn
+ ~~FAST~~ _SUPER FAST_ to use

All you need is,
1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html)
2. double-click it
3. add your tasks
4. let is manage it for you :smile:

And it is __FREE!__

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

#
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

teojunda committed:
Level-0

teojunda committed:
Created new TextTemplate class

teojunda committed:
Level-1

teojunda committed:
Level-2

teojunda committed:
Level-3

teojunda committed:
Created different types of tasks

teojunda committed:
added automated test case

teojunda committed:
Implemented error checking

teojunda committed:
implemented delete functionality

teojunda committed:
Add save feature

teojunda committed:
Merge branch 'branch-Level-7'
Add feature to save data between sessions

teojunda committed:
Add dateTime feature

teojunda committed:
Complete dateTime feature

teojunda committed:
Create Ui class

teojunda committed:
Merge branch 'add-gradle-support'
Merge gradle support

teojunda committed:
Add test case for InputParser, TaskList

teojunda committed:
Update build gradle main class

teojunda committed:
Add find keyword feature

teojunda committed:
Standardise coding standard

teojunda committed:
Add Javadocs

teojunda committed:
Resolve merge conflict

teojunda committed:
Resolve merge conflict

teojunda committed:
Enforce checkstyle

teojunda committed:
Update javafx dependencies

teojunda committed:
Create duke gui

teojunda committed:
Merge branch 'branch-Level-10'
Create duke gui

timothysashimi commented:
I don't think there's any major violations of coding standards! For eg. your package names start with a lower letter, and the method names are all in camelcase convenions :)

timothysashimi commented:
`
    public static final String LINE_BREAK = "________________________";
    public static final String GREETING = "Hello! I'm Beepo.\nWhat can I do for you?";
    public static final String EXIT = "Bye. Hope to see you again soon!";
`
static variables are also named using uppercases

choowengyan commented on src/main/java/duke/Actions.java:
```diff
@@ -0,0 +1,9 @@
+ * These actions are used to interpret user commands and execute corresponding operations.
+ */
+public enum Actions {
+    BYE, LIST, MARK, UNMARK, DELETE, TODO, DEADLINE, EVENT, INVALID, MATCH
```
I like how your choice of using enumerations for various action.

choowengyan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,90 @@
+        this.ui = new Ui();
+        this.inputParser = new InputParser();
+        this.storage = new Storage(filePath);
+        this.ready = true;
```
Perhaps better if this is `isReady` ?

choowengyan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,80 @@
+ * The Storage class handles the loading and saving of tasks to and from a file.
+ */
+public class Storage {
+    private final String filePath;
```
Perhaps it would be better to use `FILE_PATH`?

choowengyan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,80 @@
+     * @param f the file to be created
+     * @throws StorageException if an error occurs while creating the file
+     */
+    private void createFile(File f) throws StorageException {
```
Good way to name the method in this way! I noticed that this act is used elsewhere as well. good job!

choowengyan commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,153 @@
+        return t;
+    }
+
+    private ArrayList<Task> find(String keyword) {
```
Perhaps there's a better way to rename this variable instead of using `find`?

choowengyan commented on src/main/java/duke/parser/FileParser.java:
```diff
@@ -0,0 +1,67 @@
+        while (scanner.hasNext()) {
+            String s = scanner.nextLine();
+            String[] parts = s.split(separator);
+            switch (parts[0]) {
```
Good job in following the SE guide for the switch case !

choowengyan reviewed (commented):
LGTM overall! Aside from the small comments by me, it is overall well structured and organized! good job !

teojunda committed:
Add assertions

FileParser, InputParser lack assertions to check splitting of user
input. Storage lack assertion to ensure data file is created.

Unintended erros in string splitting can cause task creation error.
Storage file can be created erroneously.

Let's add some assertions to the task creation and file creation methods
in those classes.

Using assertions will ensure that the string input is split into the
correct parts, and the data file is created properly.

teojunda committed:
Add comments to improve code readability

There are not comments for method in FileParser and InputParser.

It is difficult to understand why strings are split multiple times.

Let's explain why strings a split in such a way - according to the
format of the input string.

The comments will help developers understand and edit the code if there
are changes to the app commands.

teojunda committed:
Merge pull request #1 from teojunda/branch-A-Assertions

Add assertions

teojunda committed:
Merge pull request #2 from teojunda/branch-A-CodeQuality

Add comments to improve code readability

teojunda committed:
Add DoAfter Task

teojunda committed:
correct checkstyle

teojunda committed:
fix bug for DoAfterTask checking

teojunda committed:
add Ui.png

teojunda committed:
Add user guide

teojunda committed:
Fix formatting for deadline task message

teojunda committed:
Fix grammer in user guide

teojunda committed:
update docs/Ui.png

teojunda committed:
Modify getResponse to saveData after every command
